pkg/logger/kafka/kafka.go

```diff
@@ -0,0 +1,95 @@
+package kafka
+
+import (
+	"github.com/twmb/franz-go/pkg/kgo"
+
+	"github.com/scribd/go-sdk/pkg/logger"
+)
+
+type (
+	KafkaLogger struct {
+		logger  logger.Logger
+		levelFn func() kgo.LogLevel
+	}
+)
+
+func NewKafkaLogger(l logger.Logger, opts ...Opt) *KafkaLogger {
+	kafkaLogger := &KafkaLogger{
+		logger: l,
+		levelFn: func() kgo.LogLevel {
+			return kgo.LogLevelError
+		},
+	}
+
+	for _, opt := range opts {
+		opt.apply(kafkaLogger)
+	}
+
+	return kafkaLogger
+}
+
+// Opt applies options to the kafka logger.
+type Opt interface {
+	apply(*KafkaLogger)
+}
+
+type opt struct{ fn func(*KafkaLogger) }
+
+func (o opt) apply(l *KafkaLogger) { o.fn(l) }
+
+// WithLevelFn sets a function that can dynamically change the log level.
+func WithLevelFn(fn func() kgo.LogLevel) Opt {
+	return opt{func(l *KafkaLogger) { l.levelFn = fn }}
+}
+
+// WithLevel sets a static level for the kgo.Logger Level function.
+func WithLevel(level logger.Level) Opt {
+	kgoLevel := kgo.LogLevelNone
+
+	switch level {
+	case logger.Panic, logger.Error, logger.Fatal:
+		kgoLevel = kgo.LogLevelError
+	case logger.Warn:
+		kgoLevel = kgo.LogLevelWarn
+	case logger.Info:
+		kgoLevel = kgo.LogLevelInfo
+	case logger.Trace, logger.Debug:
+		kgoLevel = kgo.LogLevelDebug
+	}
```

Comment:
I'm not sure but do we need a default case? 🤔 

```golang
default:
  kgoLevel = kgo.LogLevelDebug
```

Or is `LogLevelNone` a good default (disabled logging)?

Reply:
@fotos Default logger is set to `error` [here](https://github.com/scribd/go-sdk/pull/42/files#diff-e769714803d020600fbaa166105b2afce5ed5bcc83c1c9a181944e48cad92e13R20). The code you are referring to is to handle the unknown log level provided.

Reply:
Got it, thanks for clarifying!